webtau-rest/src/test/groovy/com/twosigma/webtau/http/HttpUrlTest.groovy

```diff
@@ -66,4 +66,28 @@ class HttpUrlTest {
             HttpUrl.extractPath('garbage://localhost:8080/relative/path').should == '/relative/path'
         } should throwException(IllegalArgumentException, 'invalid url: garbage://localhost:8080/relative/path')
     }
+
+    @Test
+    void "extract query params from a given url"() {
+        def expectedParams = [a: ['b'], c: ['d']]
```

Comment:
since extract params is a generic util function, do we want values to be defaulted to arrays?
when we do `http.get`, values you pass there are not arrays.

Reply:
Presumably you can pass them as arrays too?  How else do you send them for query params which can be repeated?

I could make it return a map of string to string or list but it'll make all the code that handles it much more horrible.

Reply:
I would pass them as an array when it is an array.
Maybe symmetry is not important here.